crates/ruff/src/rules/ruff/rules/pairwise_over_zipped.rs

```diff
@@ -0,0 +1,114 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::source_code::Stylist;
+use rustpython_parser::ast::{Expr, ExprKind};
+
+use crate::checkers::ast::Checker;
+use crate::settings::types::PythonVersion;
+use crate::Range;
+use ruff_python_ast::helpers::unparse_constant;
+
+#[violation]
+pub struct PairwiseOverZipped {}
+
+impl Violation for PairwiseOverZipped {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Prefer `itertools.pairwise()` to `zip()` if trying to fetch an iterable's successive overlapping pairs.")
+    }
+}
+
+#[derive(Debug)]
+struct SliceInfo {
+    arg_name: String,
+    slice_start: i32,
+    slice_end: i32,
+}
+
+impl SliceInfo {
+    pub fn new(arg_name: String, slice_start: i32, slice_end: i32) -> Self {
+        Self {
+            arg_name,
+            slice_start,
+            slice_end,
+        }
+    }
+}
+
+// Get arg name, lower bound, and upper bound for an expression, if it's a slice
+fn get_slice_info(expr: &Expr, stylist: &Stylist) -> Option<SliceInfo> {
+    let ExprKind::Subscript { value, slice, .. } = &expr.node else {
+        return None;
+    };
+
+    let ExprKind::Name { id: arg_id, .. } = &value.node else {
+        return None;
+    };
+
+    let mut lower_bound = 0;
+    let mut upper_bound = 0;
+    if let ExprKind::Slice { lower, upper, .. } = &slice.node {
+        if lower.is_some() {
+            if let ExprKind::Constant {
+                value: lower_value, ..
+            } = &lower.as_ref().unwrap().node
+            {
+                lower_bound = unparse_constant(lower_value, stylist)
+                    .parse::<i32>()
+                    .unwrap_or(0);
```

Comment:
We can’t just default unrecognized things to `0`. `a[0:-1]` and `a[2147483648:-1]` are quite different but this function treats them identically.